lib/bpk-component-snackbar/README.md

```diff
@@ -0,0 +1,56 @@
+# bpk-component-snackbar
+
+> Backpack React Native snackbar component.
+
+## Installation
+
+Check the main [Readme](https://github.com/skyscanner/backpack-react-native#usage) for a complete installation guide.
+
+## Usage
+
+```js
+import React from 'react';
+import { Alert } from 'react-native';
+import Buttom from 'backpack-react-native/bpk-component-buttom';
+import { icons } from 'backpack-react-native/bpk-component-icon';
+import BpkSnackbar, { LENGTH_SHORT } from 'backpack-react-native/bpk-component-snackbar';
+
+export default () => (
+  <Buttom
+    title="Show"
+    onPress={() => {
+      BpkSnackbar.show({
+        text: 'Snackbar text',
+        duration: LENGTH_SHORT,
+        title: 'Title',
+        icon: icons.flight,
+        action: {
+          text: 'Action',
+          onPress: () => {
+            Alert.alert('Action pressed');
+          }
+        }
+      })
+    }}
+  />
+);
+```
+
+## Props
+
+### BpkSnackbar
+
+| Property         | PropType                                       | Required | Default Value |
+| ---------------- | ---------------------------------------------- | -------- | ------------- |
+| `duration`       | `number`                                       | true     | `LENGTH_SHORT`|
+| `text`           | `string`                                       | true     | -             |
+| `title`          | `string`                                       | false    | -             |
+| `icon`           | `string`                                       | false    | -             |
+| `action`         | `object`                                       | false    | -             |
+| `action.text*`   | `string`                                       | true     | -             |
+| `action.icon*`   | `string`                                       | true     | -             |
+| `action.onPress` | `function`                                     | true     | -             |
+
+**NOTES**
+
+- when `action` is provided `action.text` or `action.icon` should be provided, but not both at the same time.
```

Comment:
I'm wondering if there's a less surprising way to do the API here.. 

Would there be much downside to making it three separate props, with warnings displayed if people try to use them incorrectly?

I guess it would be like:

```
onActionButtonPressed: func
actionButtonText: string
actionButtonIcon: BpkIcon

Note: When `onActionButtonPressed` is provided you must also pass in either `actionButtonText` or `actionButtonIcon`. You cannot pass both.
``` 

And then in the code throw warnings/errors if they pass in both `actionButtonText` and `actionButtonIcon`.

I realise this might be more difficult/impossible due to the bridging, so only if it's actually feasible.

Reply:
Honestly I think different props is worse as it gives the impression you can have both an action button and a text button at the same time, which is not true. 

Plus the way types are defined now flow will ensure it is impossible to set `text` and `icon` at the same time, following this proposed api I'm not sure that will be possible.

Reply:
Yeah fair enough. Let's leave it this way then